### ISSUE TITLE
Unidentifieable error running kubeone apply

### ISSUE THREAD

**Issue by danielr1996** [kubermatic/kubeone#2553]
### What happened?
I tried to deploy a k8s cluster on proxmox using the barematel provider and ubuntu jammy cloud image and got the following error(s)
```console
 kubeone apply -m kubeone/kubeone.yaml                                                                                                                         
INFO[15:35:35 CET] Determine hostname...                        
INFO[15:35:36 CET] Determine operating system...                
INFO[15:35:36 CET] Running host probes...                       
The following actions will be taken: 
Run with --verbose flag for more information.
        + initialize control plane node "k8s-c0" (172.18.0.1) using 1.24.0
        + join static worker node "k8s-w1" (172.18.0.2)

Do you want to proceed (yes/no): yes

INFO[15:35:43 CET] Determine hostname...
INFO[15:35:43 CET] Determine operating system...
INFO[15:35:43 CET] Running host probes...
INFO[15:35:43 CET] Installing prerequisites...                  
INFO[15:35:43 CET] Creating environment file...                  node=10.0.1.10 os=ubuntu
INFO[15:35:43 CET] Configuring proxy...                          node=10.0.1.10 os=ubuntu
INFO[15:35:43 CET] Installing kubeadm...                         node=10.0.1.10 os=ubuntu
INFO[15:36:18 CET] Creating environment file...                  node=10.0.1.11 os=ubuntu
INFO[15:36:18 CET] Configuring proxy...                          node=10.0.1.11 os=ubuntu
INFO[15:36:18 CET] Installing kubeadm...                         node=10.0.1.11 os=ubuntu
INFO[15:37:17 CET] Generating kubeadm config file...            
INFO[15:37:17 CET] Determining Kubernetes pause image...
INFO[15:37:17 CET] Uploading config files...                     node=10.0.1.10
INFO[15:37:17 CET] Uploading config files...                     node=10.0.1.11
INFO[15:37:18 CET] Pre-pull images                               node=10.0.1.10
INFO[15:38:13 CET] Configuring certs and etcd on control plane node... 
INFO[15:38:13 CET] Ensuring Certificates...                      node=10.0.1.10
INFO[15:38:17 CET] Downloading PKI...                           
INFO[15:38:17 CET] Creating local backup...                      node=10.0.1.10
INFO[15:38:17 CET] Uploading PKI...
INFO[15:38:17 CET] Configuring certs and etcd on consecutive control plane node...
INFO[15:38:17 CET] Initializing Kubernetes on leader...
INFO[15:38:17 CET] Running kubeadm...                            node=10.0.1.10
WARN[15:42:28 CET] Task failed, error was: runtime: running task on "10.0.1.10"
ssh: initializing kubeadm cluster
ssh: exec
ssh: popen
Process exited with status 1
set -xeuo pipefail
export "PATH=$PATH:/sbin:/usr/local/bin:/opt/bin"
if [[ -f /etc/kubernetes/admin.conf ]]; then
        sudo kubeadm  token create znuob0.yu1tvw9nl1lebus3 --ttl 1h0m0s
        exit 0;
fi
sudo kubeadm  init --config=./kubeone/cfg/master_0.yaml
stderr: + export PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/sbin:/usr/local/bin:/opt/bin
+ PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/sbin:/usr/local/bin:/opt/bin
+ [[ -f /etc/kubernetes/admin.conf ]]
+ sudo kubeadm init --config=./kubeone/cfg/master_0.yaml
W1224 14:38:17.737804    3983 initconfiguration.go:332] [config] WARNING: Ignored YAML document with GroupVersionKind kubeadm.k8s.io/v1beta3, Kind=JoinConfiguration
W1224 14:38:17.740643    3983 initconfiguration.go:120] Usage of CRI endpoints without URL scheme is deprecated and can cause kubelet errors in the future. Automatically prepending scheme "unix" to the "criSocket" with value "/run/containerd/containerd.sock". Please update your configuration!
W1224 14:38:17.740680    3983 utils.go:69] The recommended value for "clusterDNS" in "KubeletConfiguration" is: [10.96.0.10]; the provided value is: [169.254.20.10]
        [WARNING SystemVerification]: missing optional cgroups: blkio
error execution phase wait-control-plane: couldn't initialize a Kubernetes cluster
To see the stack trace of this error execute with --v=5 or higher

```

It seems that the error comes from kubeadm, but I don't really know whats going on, as this all seem to be just warnings.

### Expected behavior
I expected the cluster to be properly installed

### What KubeOne version are you using?

<details>

```console
{
  "kubeone": {
    "major": "1",
    "minor": "5",
    "gitVersion": "1.5.4",
    "gitCommit": "d814bd9007381d0eb06fefb4f7345a128b8f8666",
    "gitTreeState": "",
    "buildDate": "2022-12-12T09:37:23Z",
    "goVersion": "go1.19.4",
    "compiler": "gc",
    "platform": "linux/amd64"
  },
  "machine_controller": {
    "major": "1",
    "minor": "54",
    "gitVersion": "v1.54.2",
    "gitCommit": "",
    "gitTreeState": "",
    "buildDate": "",
    "goVersion": "",
    "compiler": "",
    "platform": "linux/amd64"
  }
}

```

</details>

### Provide your KubeOneCluster manifest here (if applicable)
```yaml
apiVersion: kubeone.k8c.io/v1beta2
kind: KubeOneCluster
name: bm-cluster
versions:
  kubernetes: '1.24.0'
cloudProvider:
  none: {}

controlPlane:
  hosts:
    - publicAddress: '10.0.1.10'
      privateAddress: '172.18.0.1'
      sshUsername: ubuntu
      sshPrivateKeyFile: '/mnt/c/Users/danie/.ssh/id_ed25519'
      taints:
        - key: "node-role.kubernetes.io/master"
          effect: "NoSchedule"
staticWorkers:
  hosts:
    - publicAddress: '10.0.1.11'
      privateAddress: '172.18.0.2'
      sshUsername: ubuntu
      sshPrivateKeyFile: '/mnt/c/Users/danie/.ssh/id_ed25519'
apiEndpoint:
  host: '10.0.0.10'
  port: 6443
machineController:
  deploy: false

```

</details>

### What cloud provider are you running

baremetal

### What operating system are you running in your cluster?
ubuntu 22.04 cloud image

**Comment by xmudrii** [comment]
As a first step, I strongly recommend solving the warning reported by kubeadm:

> [WARNING SystemVerification]: missing optional cgroups: blkio

As per [kubernetes/system-validators](https://github.com/kubernetes/system-validators/blob/737f948078e6648f34337a8aaff892f1d846bf6b/validators/types_unix.go#L71-L73), this cgroup is used by containerd:

```go
// The blkio cgroup is optional since some kernels are compiled without support for block I/O throttling.
// Containerd and cri-o will use blkio to track disk I/O and throttling in both cgroup v1 and v2.
"blkio",
```

Once you solve that:

1. Reset the cluster by running `kubeone reset -m kubeone/kubeone.yaml` -- this is going to undo most of the operations done by KubeOne (and kubeadm)
2. Run apply again but with the verbose flag: `kubeone reset -m kubeone/kubeone.yaml -v`
3. SSH to the affected instance, then:
  1. Check the status of containerd and kubelet services (`sudo systemctl status containerd` and `sudo systemctl status kubelet`)
  2. Check the kubelet logs (e.g. `journalctl -fu kubelet.service`)
  3. Check if containers are running (`sudo crictl ps -a`)

If that doesn't solve the issue, please post the complete verbose logs, kubelet logs, and output of the `crictl` command. Make sure to redact any secrets before posting logs.

**Comment by danielr1996** [comment]
Thanks for the fast answer, I will try to do a normal kubeadm install first and fix all issues first, then come back and try it with kubeone. Also your information seems really helpfull for general debugging kubeone, thanks for that. I will report my findings

**Comment by xmudrii** [comment]
@danielr1996 It seems like the blkio warning can be ignored -- I also got it on AWS with Ubuntu 22.04, but the cluster was fully functional. We'll further investigate that issue, but it's probably not the reason why KubeOne fails for you.

**Comment by danielr1996** [comment]
Thanks for the feedback, I resorted to running kubeadm myself and install k8s 1.26 as none of the warnings appeared there, maybe it's just a version issue. Will try kubeone again when it supports 1.26, then it's also easier to debug issues because I can manually run kubeadm the same way kubeone does and see where the problem might be

**Comment by xmudrii** [comment]
I researched a bit further and this is a false warning coming from kubeadm that's going to be fixed in the upcoming Kubernetes 1.25 and 1.24 patch releases (https://github.com/kubernetes/kubernetes/pull/114059 and https://github.com/kubernetes/kubernetes/pull/114060). 1.26 is not affected by this issue, so that's the reason why you don't see a warning when using 1.26.

In the meanwhile, I just noticed one thing in your KubeOneCluster manifest that I recommend checking:

```yaml
apiEndpoint:
  host: '10.0.0.10'
  port: 6443
```

I'm not sure what's on that host (`10.0.0.10`), but make sure that it's properly routing traffic to the control plane node. For example, make sure that pinging or sending traffic via that IP address is going to reach the control plane node. The host provided there can either be:

- a Load Balancer to the control plane nodes
- one of the control plane nodes (but in that case there's no real high-availability)

Considering that you have only one control plane node, you might want to just put its IP address and use it as the API endpoint.
